### ISSUE TITLE
[BUG] Message history is not always synchronized normally after a long absence

### ISSUE THREAD

**Issue by Pitstopper** [RocketChat/Rocket.Chat.Android#1353]
Your Rocket.Chat.Android version: 2.3.0 - 2.4.0
Mobile device model: Xiaomi Redmi 3
OS version: MIUI 9 (8.4.12) , Developer ROM
Your Rocket.Chat Server version: 0.65.1

Sometimes after a long absence on the channel when I run the chat program on the smartphone, I do not see a pretty large array of message history. They just do not load. This happens almost every day and does not synchronize somewhere around 1 - 1.5 hours of messages. It occurs without any apparent reason. In the web version of chat, all messages are displayed normally.
Web-version:
![img_20180612_081320](https://user-images.githubusercontent.com/18574776/41271361-ca93822c-6e18-11e8-9e1b-88d9ccc73844.jpg)

Android version:
![img_20180612_080738](https://user-images.githubusercontent.com/18574776/41271372-e6ad7c9c-6e18-11e8-8360-e8bec9dad26b.jpg)

UPD. Today it happened again.

Android version:
![img_20180614_091436](https://user-images.githubusercontent.com/18574776/41394834-2357c250-6fb4-11e8-9bc1-066e084fa719.jpg)

Web-version:

![img_20180614_091722](https://user-images.githubusercontent.com/18574776/41394851-32cdcaf4-6fb4-11e8-83d0-40c6dcfcef04.jpg)

**Comment by rafaelks** [comment]
@Pitstopper Thanks for reporting the issue. Just to be clear: if you go out of the room and open it again, will the messages appear?

**Comment by Pitstopper** [comment]
@rafaelks No. Exiting the channel, logging out / logging on to the server does not help. Helps only clearing the cache and application data with android tools.

**Comment by ged-nn** [comment]
+1 
Deleted messages remain visible, not cleared.
Helps only clearing the application data with android tools.

**Comment by luckcoolla** [comment]
@Pitstopper please check the issue on the beta branch
https://github.com/RocketChat/Rocket.Chat.Android/pull/1615

**Comment by Pitstopper** [comment]
@luckcoolla Hi!  I'm testing since the morning. I could not reproduce this bug yet. It looks like you fixed it.

**Comment by Pitstopper** [comment]
@luckcoolla Большое спасибо! :)

**Comment by Pitstopper** [comment]
Your Rocket.Chat.Android version: 2.6.1 build 2043-2b41c7d1-play
Mobile device model: Xiaomi Redmi 3
OS version: MIUI 9 (8.4.12) , Developer ROM
Your Rocket.Chat Server version: 0.69.2

I observe history synchronization errors again (

History in mobile client:

![img_20180927_093507](https://user-images.githubusercontent.com/18574776/46127281-e58d4d00-c238-11e8-8047-a9d94e8ef427.jpg)

History in web interface:

![screenshot-eaaa party-2018 09 27-09-17-20](https://user-images.githubusercontent.com/18574776/46127424-49b01100-c239-11e8-90c2-6aa88f9a0c02.png)

**Comment by luckcoolla** [comment]
Hi, could you please provide some steps before this issue?

**Comment by Pitstopper** [comment]
@luckcoolla Oh sure. What are you interested in?

**Comment by Pitstopper** [comment]
Before going to bed I turn off the wifi on the smartphone. And in the morning I turn it on, then go into the chat and watch the situation described above. I also watched this a couple of times if there was a bad quality Internet connection (for example, on 2G-3G)
